MobileBuy/sample/src/main/java/com/shopify/sample/activity/DiscountActivity.java

```diff
@@ -102,14 +109,35 @@ public void onClick(View v) {
         });
 
         Button checkoutButton = (Button) findViewById(R.id.checkout_button);
+
+        OptimizelyClient optimizelyClient = app.getOptimizelyManager().getOptimizely();
+        String userId = app.getUser();
+        String cartCTAExperimentKey = getIntent().getStringExtra("optlyCartCtaExperimentKey");
+        String userAttributesString = getIntent().getStringExtra("optlyUserAttributes");
```

Comment:
So this will in as a stringified JSON object, which I assume will work with the code below. Just wanted to make sure you knew that :) 

It'll look like this:

`"{\"thriftless\": \"true\"}"`